source/data-api.txt

```diff
@@ -52,298 +73,72 @@ The Data API supports two types of endpoints:
-          :copyable: false
-
-          {
-             "$binary": {
-                "base64": "e67803a39588be8a95731a21e27d7391",
-                "subType": "05"
-             }
-          }
-     
-     - .. code-block:: json
-          :copyable: false
+For each incoming request, the Data API:
 
-          {
-             "Subtype": 5,
-             "Data": "e67803a39588be8a95731a21e27d7391"
-          }
+1. **Authenticates the calling user.** This might involve validating an
+   access token, logging in with header credentials, or directly
+   assigning a specific runtime user based on your configuration.
 
-   * - ``Date``
-     
-     - .. code-block:: json
-          :copyable: false
+2. **Authorizes the request.** This ensures that the user sent a
+   well-formed request and has permission to perform the requested
+   operation based on your endpoint authorization scheme.
 
-          {
-             "$date": {
-                "$numberLong": "1641954803067"
-             }
-          }
-     
-     - .. code-block:: json
-          :copyable: false
+3. **Runs the requested operation.** This might involve reading or
+   writing data in Atlas with a generated endpoint or invoke a custom
+   function that you wrote.
 
-          "2022-01-12T02:33:23.067Z"
+   For requests that read or write data in Atlas, the Data API also
+   enforces the access control rules and document schemas defined in
+   your App. This means that users can only access data they're allowed
+   to read and write. Requests fail if they include an invalid write
+   operation.
 
-   * - ``Decimal128``
-     
-     - .. code-block:: json
-          :copyable: false
+4. **Returns an HTTPS response to the caller.** The response includes
+   the result of a generated endpoint operation or any data that you
+   return from a custom endpoint. Requests can choose to receive the
+   response in either :ref:`JSON or EJSON format
+   <data-api-data-formats>`.
 
-          { "$numberDecimal": "9823.1297" }
-     
-     - .. code-block:: json
-          :copyable: false
+When to Use the Data API
+------------------------
 
-          "9823.1297"
+We recommend using the Data API when:
 
-   * - ``Document``
-     
-     - .. code-block:: json
-          :copyable: false
+- You want to run MongoDB operations from a web application or other
+  client that you can't trust.
 
-          { <content> }
-     
-     - .. code-block::
-          :copyable: false
+- You can't or don't want to manage a MongoDB driver in your server-side
+  environment. For example, some edge compute environments don't support
+  database drivers or connection pooling.
 
-          { <content> }
+- You want to integrate Atlas data access into a federated API gateway
 
-   * - ``Double``
-     
-     - .. code-block:: json
-          :copyable: false
+- You want to connect to App Services from an environment not currently
+  supported by a Realm SDK and don't want to use a driver to connect
```

Comment:
ref to realm sdks landing page here? https://www.mongodb.com/docs/realm/

tho on reading the landing page, it really positions realm as a mobile DB, rather than a mobile DB + client to access App Services, so reader might be confused if they're not familiar w realm and wind up there, so 🤷 

curious your thoughts here. 

obvi out of scope for this PR, but maybe we should have a page about realm on the App Services docs explaining what Realm SDKs are, and how they related to App Services.

Reply:
> on reading the landing page, it really positions realm as a mobile DB, rather than a mobile DB + client to access App Services, so reader might be confused if they're not familiar w realm and wind up there

Link could be helpful. I think the paragraph that linked them there has enough context that some/most SDKs have an App Services integration.

> obvi out of scope for this PR, but maybe we should have a page about realm on the App Services docs explaining what Realm SDKs are, and how they related to App Services

Agreed - I thought we at least had this on the introduction page but we don't even really have it there, just a couple mentions of the SDKs in broader contexts.